### PR TITLE
call list pdf export

### PR DESCRIPTION
**Pull request by MaximilianKrambach**

### CODE REVIEW

**Comment by MaximilianKrambach** [comment]
ping @emanuelschuetze

File: client/src/app/core/ui-services/pdf-document.service.ts [hunk 1]
```diff
@@ -118,16 +126,23 @@ export class PdfDocumentService {
...
      * @param metadata
+     * @param config additional config values
      * @param imageUrls Array of optional images (url, placeholder) to be inserted
      * @returns the pdf document definition ready to export
      */
-    private async getStandardPaper(documentContent: object, metadata?: object, imageUrls?: string[]): Promise<object> {
+    private async getStandardPaper(
+        documentContent: object,
+        metadata?: object,
+        config: PdfLayoutConfigs = {},
```
Comment (tsiegleauq) [inline client/src/app/core/ui-services/pdf-document.service.ts]: `config?: PdfLayoutConfigs;`

File: client/src/app/site/motions/components/call-list/call-list.component.ts [hunk 2]
```diff
@@ -83,4 +86,11 @@ export class CallListComponent extends BaseViewComponent {
     public csvExportCallList(): void {
         this.motionCsvExport.exportCallList(this.motions);
     }
+
+    /**
+     * Trigger a pdf export of the call list
```
Comment (tsiegleauq) [inline client/src/app/site/motions/components/call-list/call-list.component.ts]: Triggers

File: client/src/app/site/motions/services/motion-pdf-export.service.ts [hunk 3]
```diff
@@ -70,4 +70,13 @@ export class MotionPdfExportService {
         };
         this.pdfDocumentService.download(doc, filename, metadata);
     }
+
+    public exportPdfCallList(motions: ViewMotion[]): void {
```
Comment (tsiegleauq) [inline client/src/app/site/motions/services/motion-pdf-export.service.ts]: missing documentation

File: client/src/app/site/motions/services/motion-pdf.service.ts [hunk 4]
```diff
@@ -413,4 +413,70 @@ export class MotionPdfService {
...
+        const table: object = {
+            table: {
+                widths: ['auto', 'auto', 'auto', '*', 'auto', 'auto'],
+                headerRows: 1,
+                body: callListTableBody.concat(callListRows)
+            }
+        };
+        return [title, table];
+    }
+    private createCallListRow(motion: ViewMotion): object {
```
Comment (tsiegleauq) [inline client/src/app/site/motions/services/motion-pdf.service.ts]: Line spacing + Comment

**Comment by MaximilianKrambach** [comment]
updated, changed the config syntax you didn't like

File: client/src/app/core/ui-services/pdf-document.service.ts [hunk 5]
```diff
@@ -247,11 +261,12 @@ export class PdfDocumentService {
             });
             this.imageUrls.push(logoHeaderRightUrl);
         }
+        const margin = [lrMargin ? lrMargin[0] : 75, 30, lrMargin ? lrMargin[0] : 75, 10];
 
         return {
             color: '#555',
             fontSize: 9,
-            margin: [75, 30, 75, 10], // [left, top, right, bottom]
+            margin: margin, // [left, top, right, bottom]
```
Comment (tsiegleauq) [inline client/src/app/core/ui-services/pdf-document.service.ts]: since this is a varibale, remove the comment

**Comment by MaximilianKrambach** [comment]
moved the comment to it's new place and rebased. @tsiegleauq

**Pull request merged**